src/rpp/rpp/subjects/replay_subject.hpp

```diff
@@ -0,0 +1,158 @@
+//                   ReactivePlusPlus library
+//
+//           Copyright Aleksey Loginov 2022 - present.
+//  Distributed under the Boost Software License, Version 1.0.
+//     (See accompanying file LICENSE_1_0.txt or copy at
+//           https://www.boost.org/LICENSE_1_0.txt)
+//
+//  Project home: https://github.com/victimsnino/ReactivePlusPlus
+
+#pragma once
+
+#include <rpp/schedulers/fwd.hpp>
+#include <rpp/subjects/fwd.hpp>
+
+#include <rpp/observers/observer.hpp>
+#include <rpp/subjects/details/base_subject.hpp>
+#include <rpp/subjects/details/subject_state.hpp>
+
+#include <list>
+#include <optional>
+
+namespace rpp::subjects::details
+{
+template<rpp::constraint::decayed_type Type>
+class replay_strategy
+{
+    struct replay_state
+    {
+        std::optional<size_t>                    count;
+        std::optional<rpp::schedulers::duration> duration;
+
+        std::list<Type>                                    values{};
+        std::list<rpp::schedulers::clock_type::time_point> time_points{};
```

Comment:
i think, it is better to union it, else it would produce x2 "new" and x2 "delete" everythime.

Reply:
done